CDTDatastore/HTTP/CDTURLSession.m

```diff
@@ -104,20 +113,26 @@ - (NSURLSessionDataTask *)createDataTaskWithRequest:(NSURLRequest *)request
 
 - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask didReceiveData:(NSData *)data
 {
-    CDTURLSessionTask *cdtURLSessionTask = [self getSessionTaskForId:dataTask.taskIdentifier];
-    data = [NSData dataWithData:data];
-    MYOnThread(self.thread, ^{
-        [cdtURLSessionTask.delegate receivedData:data];
-    });
-
+    NSMutableData * storedData = [self.dataMap objectForKey:@(dataTask.taskIdentifier)];
+    if (!storedData) {
+        storedData = [NSMutableData data];
+        [self.dataMap setObject:storedData forKey:@(dataTask.taskIdentifier)];
+    }
```

Comment:
There is a thread guarantee as long as we don't provide our own queue, comment to that effect added 5955932